ui/lib/core/addon/decorators/fetch-secrets-engine-config.ts

```diff
@@ -5,35 +5,48 @@
 
 import Route from '@ember/routing/route';
 
+import type Store from '@ember-data/store';
+import type SecretMountPath from 'vault/services/secret-mount-path';
+import type Transition from '@ember/routing/transition';
+import type Model from '@ember-data/model';
+import type AdapterError from 'ember-data/adapter'; // eslint-disable-line ember/use-ember-data-rfc-395-imports
+
 /**
- * the overview, configure, configuration and roles routes all need to be aware of the config for the engine
+ * for use in routes that need to be aware of the config for a secrets engine
  * if the user has not configured they are prompted to do so in each of the routes
  * decorate the necessary routes to perform the check in the beforeModel hook since that may change what is returned for the model
  */
 
-export function withConfig() {
-  return function decorator(SuperClass) {
+interface BaseRoute extends Route {
+  store: Store;
+  secretMountPath: SecretMountPath;
```

Comment:
Is it worth mentioning this sets the `backend` argument for the query methods?

Reply:
So if the route is typescript it will complain if you don't inject both of these services which is helpful but to go along with your previous comment I think an entry in our docs for this decorator would be very useful.